### PR TITLE
Change accumulator count dtype to int64 in tf.keras.layers.experimental.preprocessing.Normalization

### LINKED ISSUE #40016: Overflow in tf.keras.layers.experimental.preprocessing.Normalization
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes (see below for code to reproduce).
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Windows 10 10.0.18363.836
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
N/A
- TensorFlow installed from (source or binary):
Binary ([conda](https://anaconda.org/anaconda/tensorflow-gpu))
- TensorFlow version (use command below):
unknown 2.1.0
- Python version:
3.6.10
- Bazel version (if compiling from source):
N/A
- GCC/Compiler version (if compiling from source):
N/A
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

Using for a `tf.keras.layers.experimental.preprocessing.Normalization` layer `norm`, `norm.adapt(dataset)` encounters overflow warnings.

**Describe the expected behavior**

Calculate norm and standard deviation correctly.

**Standalone code to reproduce the issue**

```python
import numpy as np
import tensorflow as tf


def gen():
    for i in range(2 ** 13):
        array = np.random.random_sample(1024*1024*4).reshape(
            (1024, 1024, 4)).astype(np.float32)
        yield array * 1024 # Exacerbate the issue.

dataset = tf.data.Dataset.from_generator(
    gen, tf.float32, tf.TensorShape([1024, 1024, 4]))

dataset = dataset.batch(4)

norm = tf.keras.layers.experimental.preprocessing.Normalization()

norm.adapt(dataset)             # This ends up with RuntimeWarnings.

print(norm.mean)                  # Result is all 'inf'.
print(norm.variance)              # Result is 0.
```

**Other info / logs**

```
d:\local\envs\tf_2_1\lib\site-packages\tensorflow_core\python\keras\layers\preprocessing\normalization.py:181: RuntimeWarning: divide by zero encountered in true_divide
  ]) / combined_count
d:\local\envs\tf_2_1\lib\site-packages\tensorflow_core\python\keras\layers\preprocessing\normalization.py:190: RuntimeWarning: invalid value encountered in reduce
  variance_contribution(accumulator) for accumulator in accumulators
d:\local\envs\tf_2_1\lib\site-packages\tensorflow_core\python\keras\layers\preprocessing\normalization.py:187: RuntimeWarning: overflow encountered in square
  accumulator.variance + np.square(accumulator.mean - combined_mean))
d:\local\envs\tf_2_1\lib\site-packages\tensorflow_core\python\keras\layers\preprocessing\normalization.py:187: RuntimeWarning: invalid value encountered in multiply
  accumulator.variance + np.square(accumulator.mean - combined_mean))
<tf.Variable 'mean:0' shape=(4,) dtype=float32, numpy=array([inf, inf, inf, inf], dtype=float32)>
<tf.Variable 'variance:0' shape=(4,) dtype=float32, numpy=array([0., 0., 0., 0.], dtype=float32)>
```

The count overflow problem could potentially be mitigated by changing the dtype [here](https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/layers/preprocessing/normalization.py#L158) to int64,

### PR DESCRIPTION
**Pull request by gfkeith**
This fixes #40016. The dtype for the [outward-facing count](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/preprocessing/normalization.py#L116) is already int64.

### CODE REVIEW

**Review by kyscg — APPROVED** [review]
Good catch, LGTM!

**Review by tanzhenyu — APPROVED** [review]
Thank you!

**Review by rthadur — APPROVED** [review]
